roiextractors/imagingextractor.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from spikeextractors.baseextractor import BaseExtractor
+from .extraction_tools import ArrayType, PathType, NumpyArray, DtypeType
+
+
+class ImagingExtractor(ABC, BaseExtractor):
+    '''An abstract class that contains all the meta-data and input data from
+       the imaging data.
+    '''
+    def __init__(self):
+        BaseExtractor.__init__(self)
+        self._memmapped = False
+
+    @abstractmethod
+    def get_frame(self, frame_idx: int, channel: int = 0) -> NumpyArray:
+        pass
+
+    @abstractmethod
+    def get_frames(self, frame_idxs: ArrayType, channel: int = 0) -> NumpyArray:
+        pass
```

Comment:
Do you think we need both of these functions? Could `get_frames` take a single frame? Or do you want to maintain strong typing?